internal/controllers/servicebinding_controller.go

```diff
@@ -50,11 +52,41 @@ type ServiceBindingReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *ServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx)
 
-	// your logic here
+	servicebinding := &servicebindingv1beta1.ServiceBinding{}
 
-	return ctrl.Result{}, nil
+	if err := r.Get(ctx, req.NamespacedName, servicebinding); err != nil {
+		if errors.IsNotFound(err) {
+			// We got a request for reconciliation, but no object exists.  Do
+			// nothing and don't requeue.
+			log.Info("Service Binding not found; ignoring.", "name", req.NamespacedName, "err", err)
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Unable to retrieve service binding", "name", req.NamespacedName)
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
Create a DeepCopy of the ServiceBinding before mutating to avoid informer cache pollution.

We can also apply the default values to the copy.


```suggestion
	servicebinding = servicebinding.DeepCopy()
	servicebinding.Default()
```